hnswlib/hnswalg.h

```diff
@@ -484,6 +484,8 @@ namespace hnswlib {
 
 
         std::priority_queue<std::pair<dist_t, tableint>> searchKnnInternal(void *query_data, int k) {
+            std::priority_queue<std::pair<dist_t, tableint  >> top_candidates;
+            if (cur_element_count == 0) return top_candidates;
```

Comment:
```suggestion
            if (cur_element_count == 0) {
                  std::priority_queue<std::pair<dist_t, tableint  >> top_candidates;
                  return top_candidates;
            }
```

Reply:
Not sure if the compiler does not optimize this, but there is no need to create `top_candidates` if it does not branch.